pkg/karmadactl/cmdinit/karmada/rbac.go

```diff
@@ -123,3 +124,119 @@ func grantAccessPermissionToAgent(clientSet kubernetes.Interface) error {
 
 	return nil
 }
+
+// grantKarmadaPermissionToAdminClusterRole grants admin clusterrole with karamda resource permission to manage karmada resources
+func grantKarmadaPermissionToAdminClusterRole(clientSet kubernetes.Interface) error {
+	annotations := map[string]string{
+		// refer to https://kubernetes.io/docs/reference/access-authn-authz/rbac/#auto-reconciliation
+		// and https://kubernetes.io/docs/reference/access-authn-authz/rbac/#kubectl-auth-reconcile
+		"rbac.authorization.kubernetes.io/autoupdate": "true",
+	}
+	labels := map[string]string{
+		// refer to https://kubernetes.io/docs/reference/access-authn-authz/rbac/#default-roles-and-role-bindings
+		"kubernetes.io/bootstrapping": "rbac-defaults",
+		// used to aggregate rules to admin clusterrole
+		"rbac.authorization.k8s.io/aggregate-to-admin": "true",
+	}
+	clusterRole := utils.ClusterRoleFromRules(adminClusterRole, []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{"autoscaling.karmada.io"},
+			Resources: []string{
+				"cronfederatedhpas",
+				"cronfederatedhpas/status",
+				"federatedhpas",
+				"federatedhpas/status",
+			},
+			Verbs: []string{
+				"get",
```

Comment:
Can we use `*`?

Reply:
The https://kubernetes.io/docs/reference/access-authn-authz/rbac/#referring-to-resources says:
>Caution: Using wildcards in resource and verb entries could result in overly permissive access being granted to sensitive resources. For instance, if a new resource type is added, or a new subresource is added, or a new custom verb is checked, the wildcard entry automatically grants access, which may be undesirable. The [principle of least privilege](https://kubernetes.io/docs/concepts/security/rbac-good-practices/#least-privilege) should be employed, using specific resources and verbs to ensure only the permissions required for the workload to function correctly are applied.

I wonder if there are other hidden trouble with wildcards :)

Reply:
I believe it is acceptable to use wildcards in specific API groups(like, autoscaling.karmada.io).

cc @XiShanYongYe-Chang

Reply:
Hi @zhy76, We actually added two clusterRoles: `karmada-edit` and `karmada-view`. These two clusterRoles have separate read and write permissions, so they need to be written separately.

Reply:
Oh,thanks,I will fix it.